restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Conference.java

```diff
@@ -553,6 +581,26 @@ private void onStopRecording(StopRecording message, ActorRef self, ActorRef send
         }
     }
 
+    private void onReceiveTimeout(ReceiveTimeout message, ActorRef self, ActorRef sender) throws Exception {
+        if (logger.isInfoEnabled()) {
+            logger.info("Conference Received Timeout, will stop conference now.");
+        }
+        onStopConference(new StopConference(SUPER_ADMIN_ACCOUNT_SID), self, sender);
+    }
+
+    private void onDownloaderResponse(DownloaderResponse message, ActorRef self, ActorRef sender) {
+        //make sure we only remove sender that is callApiClient and was created by this conference actor
+        //so we don't kill a sender that has accidently broadcast a DownloaderResponse to this conference.
+        if(callApiClients.contains(sender)){
```

Comment:
@maria-farooq as said earlier no need to keep a list of `CallApiClient` and then check if sender is on the list.
I believe we should stop sender straight away UNLESS if there is a chance to receive DownloaderResponse from other actors, which I cannot think of such a case.

Even if that's the case (other actors sends DownloaderResponse to Conference) maybe its better to create another wrapper message for the responses of `CallApiClient`